src/Functions/array/arrayAggregation.cpp

```diff
@@ -152,9 +152,77 @@ struct ArrayAggregateImpl
         return result;
     }
 
+    template <AggregateOperation op = aggregate_operation>
+    requires(op == AggregateOperation::min || op == AggregateOperation::max)
+    static bool executeMinOrMax(const ColumnPtr & mapped, const ColumnArray::Offsets & offsets, ColumnPtr & res_ptr)
```

Comment:
we never return false

Reply:
+1 Initially I thought there was some cases where it might be impossible to implement it, but since all types implement getPermutation and compareAt it never fails AFAICT. It's the same as `ORDER BY column::{type}`. I'll change it